### PR TITLE
/dev/random bug fixes

### PR DESCRIPTION
**Pull request by n8sh**
If there were multiple `/dev/random` CSPRG instances, one of them being destroyed broke the others. `getBytes` also didn't check the number of bytes read from `/dev/random`.

### CODE REVIEW

**Comment by JonathanWilbur** [comment]
So you don't think I'm a total dummy, my rationale for making the `randomFile` member `static` was so that it would not be opened multiple times; there would just be one `randomFile` instance shared across all `CSPRNG`s, and when each `CSPRNG` fell out of scope, it would just release its reference to the pointer with `File.detach()` instead of `File.close()`. Unfortunately, as I learned this morning, `File` does not count references the way I thought it did. The whole time, there was only one reference to the open file, no matter how many `CSPRNG`s were open, and it would get closed when the destructor was called.

When I did testing of this--which you can find in `test/multiple.d`--I just ensured that a `CSPRNG` fell out of scope, rather than explicitly calling `.destroy()`. Apparently just falling out of scope does not call the destructor immediately or something. 

Having said that, **these are good pull requests. I'll accept them.** But, I would like to somewhat revert the code to using a static `randomFile`. I experimented with managing open references to `CSPRNG` just using an internal `static size_t openInstances` counter, which would increment after successfully calling a constructor, and decrement after successfully calling the destructor. Do you have any objections to me doing it this way? Is there a reason I should have separate `randomFile` instances for each instance of `CSPRNG`?

**Review by JonathanWilbur — APPROVED** [review]
Very clever! I didn't even think about the fact that /dev/random could return fewer bytes than were in the buffer.

**Review by JonathanWilbur — APPROVED** [review]
Please see my comment on this PR.

**Comment by n8sh** [comment]
>But, I would like to somewhat revert the code to using a static `randomFile`

I agree that would be best, using the approach you describe.

**Comment by n8sh** [comment]
>Apparently just falling out of scope does not call the destructor immediately or something.

BTW, it does for structs but not for classes.
